src/rules/npmNamingRule.ts

```diff
@@ -1,25 +1,177 @@
+                            type: "string",
+                            enum: [Mode.NameOnly],
+                        },
+                        "single-line": {
+                            description: "Whether to print error messages in a single line. Used for testing.",
+                            type: "boolean",
+                        },
+                        "required": ["mode"],
+                    },
+                },
+                {
+                    type: "object",
+                    properties: {
+                        "mode": {
+                            type: "string",
+                            enum: [Mode.Code],
+                        },
+                        "errors": {
+                            type: "array",
+                            items: {
+                                type: "array",
+                                items: [
+                                    {   description: "Name of the check.",
+                                        type: "string",
+                                        enum: [ErrorKind.NeedsExportEquals, ErrorKind.NoDefaultExport] as ExportErrorKind[],
+                                    },
+                                    {
+                                        description: "Whether the check is enabled or disabled.",
+                                        type: "boolean",
+                                    },
+                                ],
+                                minItems: 2,
+                                maxItems: 2,
+                            },
+                        },
+                        "single-line": {
+                            description: "Whether to print error messages in a single line. Used for testing.",
+                            type: "boolean",
+                        },
+                        "required": ["mode"],
+                    },
+                },
+            ],
+        },
+        optionExamples: [
+            true,
+            [true, { mode: Mode.NameOnly }],
+            [true, { mode: Mode.Code, errors: [[ErrorKind.NeedsExportEquals, true], [ErrorKind.NoDefaultExport, false]] }],
+        ],
         type: "functionality",
         typescriptOnly: true,
     };
 
     apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk);
+        return this.applyWithFunction(sourceFile, walk, parseOptions(this.ruleArguments));
+    }
+}
+
+function parseOptions(args: unknown[]): Options {
+    if (args.length === 0) {
+        return defaultOptions;
+    }
+
+    const arg = args[0] as { [prop: string]: unknown } | null | undefined;
+    if (arg == null) {
+        return defaultOptions;
+    }
+
+    if (!arg.mode || typeof arg.mode !== "string") {
+        return defaultOptions;
+    }
+
+    const mode = parseMode(arg.mode);
+    if (!mode) {
+        return defaultOptions;
+    }
+
+    let singleLine;
```

Comment:
```suggestion
    const singleLine = arg["single-line"] ? true : undefined;
```

or maybe just `!!arg["single-line"]`, I don't know if the `undefined` is ever important.